weed/storage/volume.go

```diff
@@ -46,6 +46,57 @@ func loadVolumeWithoutIndex(dirname string, collection string, id VolumeId, need
 	e = v.load(false, false, needleMapKind)
 	return
 }
+func verifyIndexFileIntegrity(indexFile *os.File) (indexSize int64, err error) {
+	var fi os.FileInfo
+	if fi, err = indexFile.Stat(); err != nil {
+		return
+	} else if indexSize = fi.Size(); indexSize != 0 && indexSize%16 != 0 {
```

Comment:
I agree with point 1 and point 3. 
Regarding point 2, I guess you mean I could use the returned indexSize to do the comparation work at outside function(assuming there is a empty idx and dat file which does not need this data integrity checking)